toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -72,9 +72,6 @@ func (s *SystemConfig) IsValid() (err error) {
 		return fmt.Errorf("invalid [BootType]: %s. Expecting values of either 'efi', 'legacy', 'none' or empty string", s.BootType)
 	}
 
-	if len(s.PackageLists) == 0 && len(s.Packages) == 0 {
```

Comment:
Out of interest, why was this check removed?

Reply:
Image definitions expect to always have packages to install (whether it's for a vhd(x) build or an iso build).
When mic uses IsoMaker to generate an iso, there are really no packages to include in the generated iso (just the squashfs). 
IsoMaker interface is an image definition file which goes through this code for validation. I'm relaxing the check so that image definitions from mic->isomaker can go through.

Reply:
Should this check be added back in the imager tool?

Reply:
Judging by the test, this check is also verifying the proper content of the image config JSON. If the `imageconfigvalidator` tool is always ran when we build images, then yes, let's move the check **and the test** there as part of this PR. However, if the validator is not part of the image build, I think we should leave the check here, since a proper image configuration expect a package list.

@gmileka, could you elaborate more on how MIC is making use of the ISO maker? It sounds like it's outside the regular flow of how the toolkit is used. I'm worried we're making changes for a custom, non-official scenario, while breaking the official behaviour.

Reply:
@PawelWMS - sure...
- IsoMaker takes a configuration file, and a pointer to rpms and user artifacts and creates an iso with all of those artifacts.
- Mic takes an existing image (vhd(x)/qcow), extracts artifacts (rootfs, etc), and creates an iso with all of those artifacts.

The main change in this PR is about allowing IsoMaker to:
- accept an image configuration that might not have the same attributes as what's required today.
- accept an image configuration through the code and by reading it from a config file and parsing it (since MIC will generate that image configuration at run time).

The change should not change/break the existing behavior in anyway. It allows different entry points into the code though + relaxied config schema requirements.

The way I see it is that there is an input front-end and an "engine" to create the ISO. This change is providing another input front-end and leverages the same "engine" to create the ISO.

Reply:
I have decided to take it out because, today, IsoMaker itself does not run the full config validation and hence I do not have to change it right now. However, there is still an issue about who validates the schema and whether we should make changes to the scheme.